From 98aba92ebb0720abfb4d676bdf29944decc6f7bf Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 21 Jun 2012 10:16:31 +1000
Subject: [PATCH] symbols: ossmath is CTRL+ALT, not FOUR_LEVEL (#43541)

having KPMU defined as FOUR_LEVEL, with 4 symbols only, triggers an xkb
error when the keypad stuff picks up the CTRL+ALT (from x11) and waits for 5
symbols instead.

X.Org Bug 43541 <http://bugs.freedesktop.org/show_bug.cgi?id=43541>

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 symbols/keypad |   11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/symbols/keypad b/symbols/keypad
index 1bab391..b362ebc 100644
--- a/symbols/keypad
+++ b/symbols/keypad
@@ -267,13 +267,12 @@ xkb_symbols "legacymath" {
 partial keypad_keys
 xkb_symbols "ossmath" {
 
-    key.type[Group1]="FOUR_LEVEL_X" ;
-
-    key <KPDV> { [   KP_Divide, 0x1002215, 0x10000F7,     XF86_Ungrab ] }; // / ∕ ÷     <XF86_Ungrab>
-    key <KPMU> { [ KP_Multiply, 0x10022C5, 0x10000D7,  XF86_ClearGrab ] }; // * ⋅ ×  <XF86_ClearGrab>
-    key <KPSU> { [ KP_Subtract, 0x1002212, 0x1002212, XF86_Prev_VMode ] }; // - − − <XF86_Prev_VMode>
+    key.type[Group1]="CTRL+ALT" ;
 
-    key <KPAD> { [      KP_Add, 0x100002B, 0x100002B, XF86_Next_VMode ] }; // + + + <XF86_Next_VMode>
+    key <KPDV> { [   KP_Divide, 0x1002215, 0x10000F7, VoidSymbol,     XF86_Ungrab ] }; // / ∕ ÷     <XF86_Ungrab>
+    key <KPMU> { [ KP_Multiply, 0x10022C5, 0x10000D7, VoidSymbol,  XF86_ClearGrab ] }; // * ⋅ ×  <XF86_ClearGrab>
+    key <KPSU> { [ KP_Subtract, 0x1002212, 0x1002212, VoidSymbol, XF86_Prev_VMode ] }; // - − − <XF86_Prev_VMode>
+    key <KPAD> { [      KP_Add, 0x100002B, 0x100002B, VoidSymbol, XF86_Next_VMode ] }; // + + + <XF86_Next_VMode>
 
 };
 
-- 
1.7.10.2

